### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by kab00kie** [sensepost/ruler#4]
Hello

Firstly, thanks for this tool!  

Second, I'm trying to test it against my work mailbox (I have permission).  I have been successful in doing it manually but using ruler I am running into a problem detailed below.  From home I am able to upload a malicious rule to my work Outlook profile, send the trigger, and receive a session from my work laptop.

I read through the currently closed issues and set up a dummy Outlook.com account for testing which appears to work no problem:

./ruler -user "XXXX" -pass XXXX -email "XXXX@outlook.com" -display -url https://autodiscover-s.outlook.com/autodiscover/autodiscover.xml
[*] Retrieving MAPI/HTTP info
[*] Doing Autodiscover for domain
[+] MAPI URL found:  https://outlook.office365.com/mapi/emsmdb/?MailboxId=00034001-fb6a-3d3c-0000-000000000000@outlook.com
[+] User DN:  /o=First Organization/ou=Exchange Administrative Group(FYDIBOHF23SPDLT)/cn=Recipients/cn=00034001FB6A3D3C
[*] Got Context, Doing ROPLogin
[*] And we are authenticated
[+] Mailbox GUID: fb29132691adc3449f1ff6738b215ba4
[*] Openning the Inbox
[+] Retrieving Rules
[+] Found 0 rules
[*] And disconnecting from server

Work:

./ruler -user XXXX -pass XXXX -email XXXX@domain.com -domain domain.com -display
[*] Retrieving MAPI/HTTP info
[*] Doing Autodiscover for domain
[x] No MAPI URL found. Trying RPC/HTTP
[*] Retrieving RPC/HTTP info
     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
[+] MAPI URL found:  
Post : Get : unsupported protocol scheme ""
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x38 pc=0x4a9402]

goroutine 1 [running]:
panic(0x815a80, 0xc82000e130)
	/usr/lib/golang/src/runtime/panic.go:481 +0x3e6
github.com/sensepost/ruler/mapi.Authenticate(0x7ffc202a765e, 0x0, 0x0)
	/home/staaldraad/Go/gopath/src/github.com/sensepost/ruler/mapi/mapi.go:199 +0x2f2
main.main()
	/home/staaldraad/Go/gopath/src/github.com/sensepost/ruler/ruler.go:166 +0x1141

Further to this, if I run with -check or -check -insecure, I get this:

[+] MAPI URL found:  
[+] Authentication succeeded and MAPI/HTTP is available

Can you offer any insight?  I can accept that maybe the tool will not work against my work, which is great, but the error messages I'm seeing seem strange.  Maybe "Post : Get : unsupported protocol scheme """ is the indicator that it won't work but the golang errors made me question.

Thanks!

**Comment by staaldraad** [comment]
Hi! Thanks for trying it out 

This is an easy one to solve, the issue lies in this

> [x] No MAPI URL found. Trying RPC/HTTP
   [*] Retrieving RPC/HTTP info

You've hit a code path that I started developing and haven't finished. When Ruler was released, it only supported MAPI/HTTP, I've been trying to get RPC/HTTP working and unfortunately I forgot to disable it before pushing some of the other changes to the master branch.

This feature currently doesn't work, and this means Ruler doesn't work against your workplace, which is great news (for you).

I'm going to disable this code branch now, thanks for letting me know!
